feature/system/tests/system_base_test/g_protocol_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package system_base_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/openconfig/ondatra"
+	"google.golang.org/grpc"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	spb "github.com/openconfig/gnoi/system"
+	authzpb "github.com/openconfig/gnsi/authz"
+	gribipb "github.com/openconfig/gribi/v1/proto/service"
+	p4rtpb "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+// TestGNMIClient validates that the DUT listens on standard gNMI Port.
+func TestGNMIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	addr := fmt.Sprintf("%s:9339", dut.Name())
```

Comment:
Is using `dut.Name()` as a valid DNS name something that we can rely on? I think in KNE environments we don't have anything that would create such a resolver entry. Should we rather use `dut.RawAPIs` or a call that gets the IP of the assigned device here? (Although I expect that `RawAPIs` then introduces new layers that are being dialed through, but I don't remember fully.)